.github/workflows/code-qa.yml

```diff
@@ -1,12 +1,14 @@
 # SPDX-FileCopyrightText: Copyright DB Netz AG and the capellambse contributors
 # SPDX-License-Identifier: CC0-1.0
 
-name: Build
+name: Code quality assurance
 
 on:
+  workflow_dispatch:
   push:
-    branches: ["*"]
+    branches: [master]
     tags: ["v*.*.*"]
+  pull_request:
```

Comment:
Do we want to keep the `push` event trigger unbounded _in addition to_ `pull_request`, as it is done in [capella-collab-manager](https://github.com/DSD-DBS/capella-collab-manager/blob/main/.github/workflows/lint.yml)? This causes the workflow to run twice for every push to an open PR, however it would make it easier to try out experimental changes in the CI (which is probably most interesting for unit tests on other OSes). As a middle ground between those, I have added the `workflow_dispatch` trigger to the code-qa workflow, which allows to manually run a workflow on demand.